### PR TITLE
Fixes for Bokeh path plots

### PR DESCRIPTION
**Pull request by philippjfr**
Fixes for hover tools on polygons and adding batching for Path plots. This PR is based off https://github.com/ioam/holoviews/pull/762 and should be merged after that one.

### CODE REVIEW

**Comment by philippjfr** [comment]
Just rebased the PR, now ready to merge.

File: holoviews/plotting/bokeh/path.py [hunk 1]
```diff
@@ -17,14 +17,34 @@ class PathPlot(ElementPlot):
...
+            eldata, elmapping = self.get_data(el, ranges, empty)
+            for k, eld in eldata.items():
+                data[k].extend(eld)
+            zorder = self.get_zorder(element, key, el)
+            val = style[zorder].get('color')
+            elmapping['color'] = 'color'
+            if isinstance(val, tuple):
+                val = rgb2hex(val)
+            data['color'] += [val for _ in range(len(eldata['xs']))]
+        if len(set(data.get('color'))) == 1:
```
Comment (jlstevens) [inline holoviews/plotting/bokeh/path.py]: I'm a little confused by this bit. In the loop, you populate ``data`` with ``'color'`` but here you pop it off again?

**Comment by jlstevens** [comment]
Other than one comment (which is only asking for clarification) this looks fine and I am happy to merge.

File: holoviews/plotting/bokeh/path.py (hunk 1 above)
Comment (philippjfr) [inline holoviews/plotting/bokeh/path.py]: Yes, if all the paths are the same color there's no need to set it individually for each path. It'll simply be set once separately from the data.

Comment (jlstevens) [inline holoviews/plotting/bokeh/path.py]: Ok, thanks for clarifying!

Comment (philippjfr) [inline holoviews/plotting/bokeh/path.py]: I'll get into the habit of adding more comments for non-obvious things like this.

**Pull request merged**